server/bleep/src/lib.rs

```diff
@@ -103,15 +102,17 @@ impl Application {
         }
 
         let config = Arc::new(config);
-        let semantic =
-            match Semantic::new(&config.model_dir, &config.qdrant_url, Arc::clone(&config)).await {
-                Ok(semantic) => Some(semantic),
-                Err(SemanticError::QdrantInitializationError) => {
-                    warn!("Qdrant initialization failed. Starting without semantic search...");
-                    None
+        let semantic = match config.qdrant_url {
+            Some(ref url) => {
+                match Semantic::initialize(&config.model_dir, url, Arc::clone(&config)).await {
+                    Ok(semantic) => Some(semantic),
+                    Err(e) => {
+                        bail!("Qdrant initialization failed: {}", e);
+                    }
                 }
-                Err(e) => bail!(e),
-            };
+            }
+            None => None,
```

Comment:
We should log here if Qdrant isn't initialised, but bloop startup continues.

Reply:
Yeah, simply working without semantic search without so much as a warning would be bad UX.